contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -536,7 +540,7 @@ func (s *SchedulerServer) getDriver() (driver bindings.SchedulerDriver) {
 }
 
 func (s *SchedulerServer) Run(hks hyperkube.Interface, _ []string) error {
-	if n := len(s.mesosRoles); n == 0 || n > 2 || (n == 2 && s.mesosRoles[0] != "*" && s.mesosRoles[1] != "*") {
+	if n := len(s.frameworkRoles); n == 0 || n > 2 || (n == 2 && s.frameworkRoles[0] != "*" && s.frameworkRoles[1] != "*") {
```

Comment:
I completely understand the lack of specific harm to the system. My concern
is related to usability: we offer two flags that affect role usage and it's
not clear to me that we've set up an operator for success. What's the value
of allowing defaultPodRoles > frameworkRoles? How is this not confusing to
an operator that doesn't understand our implementation at the code-level?

On Tue, Dec 8, 2015 at 11:26 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/scheduler/service/service.go
> <https://github.com/kubernetes/kubernetes/pull/18348#discussion_r46975645>
> :
>
> > @@ -536,7 +540,7 @@ func (s *SchedulerServer) getDriver() (driver bindings.SchedulerDriver) {
> >  }
> >
> >  func (s *SchedulerServer) Run(hks hyperkube.Interface, _ []string) error {
> > -	if n := len(s.mesosRoles); n == 0 || n > 2 || (n == 2 && s.mesosRoles[0] != "*" && s.mesosRoles[1] != "*") {
> > +	if n := len(s.frameworkRoles); n == 0 || n > 2 || (n == 2 && s.frameworkRoles[0] != "*" && s.frameworkRoles[1] != "*") {
>
> It does not harm if defaultPodRoles is too big.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18348/files#r46975645>.
>